src/Catrobat/Controller/Web/ProgramController.php

```diff
@@ -646,6 +649,46 @@ private function checkReportedByUser(Program $program, ?User $user): bool
     return $isReportedByUser;
   }
 
+  private function projectCustomTranslationGetAction(Request $request, string $id): Response
+  {
+    $project = $this->program_manager->find($id);
+    try {
+      if (!$this->program_manager->isProjectVisibleForCurrentUser($project)) {
+        return Response::create(null, Response::HTTP_NOT_FOUND);
+      }
+    } catch (NoResultException $e) {
+      return Response::create(null, Response::HTTP_NOT_FOUND);
+    }
+
+    if (!$request->query->has('field')
+      || !$request->query->has('language')) {
+      return Response::create(null, Response::HTTP_BAD_REQUEST);
+    }
+
+    $field = $request->query->get('field');
+    $language = $request->query->get('language');
+
+    try {
+      switch ($field) {
+        case 'name':
+          $result = $this->translation_delegate->getProjectNameCustomTranslation($project, $language);
+          break;
+        case 'description':
+          $result = $this->translation_delegate->getProjectDescriptionCustomTranslation($project, $language);
+          break;
+        case 'credit':
+          $result = $this->translation_delegate->getProjectCreditCustomTranslation($project, $language);
+          break;
+        default:
+          return Response::create(null, Response::HTTP_BAD_REQUEST);
```

Comment:
Should there be an error message here saying that the field is not valid?

Reply:
i don't think so, because this error should never reach the user